### DISCUSSION TITLE
How to generate Allure Report programmatically - after end of test suite

### DISCUSSION THREAD

**Discussion by kamil-nowocin** [allure-framework/allure2#1304] [Questions & Support] [answered]
How to generate Allure report from Java code?
Right now I'm using allure-comandline and generate report via `allure generate build/allure-results --clean`
I'd like to automatically generate it after end of test suite. I did some research and found this https://mvnrepository.com/artifact/io.qameta.allure/allure-generator but unfortunately this doesn't work well or I don't know how to use it properly.
Any tips? Thanks!

**Comment by ezlo-maksymk** [comment]
Hi Kamil,

you can use maven (hope gradle)

with maven it will be 
pom.xml  
```
<project>
.......
<build>
        <plugins>
            <plugin>
                <groupId>io.qameta.allure</groupId>
                <artifactId>allure-maven</artifactId>
                <version>2.10.0</version>
                <configuration>
                    <reportVersion>2.10.0</reportVersion>
                </configuration>
            </plugin>
        </plugins>
    </build>

</project>
```

and than in terminal in the folder of your project run 
`mvn allure:serve`

**Reply by kamil-nowocin** [reply]
I don't want to generate it from terminal. I already know how to do that.
As title of this topic - I want to do it via java code.

**Reply by kamil-nowocin** [reply]
And tbh I don't really want to execute terminal command via java as ` Process p = Runtime.getRuntime().exec(command);`

**Reply by baev** [reply]
https://github.com/allure-framework/allure2/blob/master/allure-generator/src/main/java/io/qameta/allure/ReportGenerator.java

Allure command line simply calls it, see https://github.com/allure-framework/allure2/blob/master/allure-commandline/src/main/java/io/qameta/allure/Commands.java#L103

Note, that you'll need to add some plugins by yourself https://github.com/allure-framework/allure2/tree/master/plugins
plugins are available in maven central (an example of loading plugins https://github.com/allure-framework/allure2/blob/master/allure-commandline/src/main/java/io/qameta/allure/Commands.java#L182-L196 ) 

And the complete example that we use for development https://github.com/allure-framework/allure2/blob/master/allure-generator/src/main/java/io/qameta/allure/DummyReportGenerator.java

**Reply by kamil-nowocin** [reply]
Thanks. Lots of information here. Hopefully I'll be able to do it by myself.

**Reply by sabag** [reply]
ok, i was able to generate using the example from DummyReportGenerator, thank you

**Comment by ezlo-maksymk** [comment]
if you will run `mvn clean install && mvn allure:serve` - it will run your tests and after that will generate report, like what you asked `I'd like to automatically generate it after end of test suite.` and will save your time to write code of allure generation and supporting it

**Reply by kamil-nowocin** [reply]
Oh, this is great, but I'm using gradle so this doesn't really help me. Anyway, thank you for your input

**Reply by baev** [reply]
there is allure gradle plugin available, so you can use it to generate report automatically after tests run

**Reply by gauravr3** [reply]
@baev what is the gradle plugin which I can generate report automatically after test finish

**Reply by PavelMal** [reply, marked as answer]
@gauravr3 
https://github.com/allure-framework/allure-gradle - you are welcome, but pay attention:
**allure-gradle 2.9+ requires Gradle 5.0+**
**allure-gradle 2.11+ requires Gradle 6.0+**

Groovy DSL:

```
plugins {
    id'io.qameta.allure' version '<latest>'
}

repositories {
    // Repository is needed for downloading allure-commandline for building the report
    mavenCentral()
}
```

**Comment by jstefek** [comment]
for those who still need to generate report programmatically and see this in 2025,  with version 2.32.0 of `allure-generator` one can simply use `new ReportGenerator(ConfigurationBuilder.bundled().build()).generate(destination, source)` 

dependency:
```
            <dependency>
                <groupId>io.qameta.allure</groupId>
                <artifactId>allure-generator</artifactId>
                <version>2.32.0</version>
            </dependency>
```
